app/docker/networks/queries.ts

```diff
@@ -0,0 +1,45 @@
+import { useQuery, useMutation, useQueryClient } from 'react-query';
+
+import { EnvironmentId } from '@/portainer/environments/types';
+import {
+  error as notifyError,
+  success as notifySuccess,
+} from '@/portainer/services/notifications';
+
+import { getNetwork, removeNetwork } from './network.service';
+import { NetworkId } from './types';
+
+export function useNetwork(environmentId: EnvironmentId, networkId: NetworkId) {
+  return useQuery(
+    ['environmentId', environmentId, 'docker', 'network', networkId],
+    () => getNetwork(environmentId, networkId),
+    {
+      onError: (err) => {
+        // if there's an error fetching the network, tell the user
+        notifyError('Failure', err as Error, 'Unable to get network');
+      },
+    }
+  );
+}
+
+export function UseDeleteNetwork(
+  environmentId: EnvironmentId,
+  networkId: NetworkId
```

Comment:
I think the mutation shouldn't take the env/networkId. it will get them when doing `mutation.mutate({envId, networkId})`

Reply:
since the mutate function takes these values, you don't need this.
```
export function useDeleteNetwork() {

```